lib/donation_system/paypal_wrapper/payment_creator.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'paypal-sdk-rest'
+
+require_relative 'creator_data_validator'
+require_relative 'creator_fields_generator'
+require_relative '../result'
+
+module DonationSystem
+  module PaypalWrapper
+    class PaymentCreator
+      include PayPal::SDK::REST
+
+      def self.execute(data)
+        new(data).execute
+      end
+
+      def initialize(data)
+        @data = data
+        @errors = []
+        config_paypal
+      end
+
+      def execute
+        Result.new(payment, errors)
+      end
+
+      private
+
+      attr_reader :data, :errors
+
+      def payment
+        return failure unless validation.okay?
+        payment = PayPal::SDK::REST::DataTypes::Payment.new(fields)
+        rescue_errors { payment.create }
+        save_error(payment.error) if payment.error
+        payment
+      end
+
+      def rescue_errors
+        yield
+      rescue PayPal::SDK::Core::Exceptions::UnauthorizedAccess
+        errors << :unauthorized_access
```

Comment:
For the future: I think the new errors object should also store both the exception error message and object.